src/metadata/iceberg_column_definition.cpp

```diff
@@ -172,4 +172,32 @@ unique_ptr<IcebergColumnDefinition> IcebergColumnDefinition::ParseStructField(re
 	                 field.has_initial_default ? &field.initial_default : nullptr);
 }
 
+bool IcebergColumnDefinition::IsPrimitiveType() {
+	switch (type.id()) {
+	case LogicalTypeId::TINYINT:
+	case LogicalTypeId::UTINYINT:
+	case LogicalTypeId::SMALLINT:
+	case LogicalTypeId::USMALLINT:
+	case LogicalTypeId::INTEGER:
+	case LogicalTypeId::BOOLEAN:
+	case LogicalTypeId::VARCHAR:
+	case LogicalTypeId::DATE:
+	case LogicalTypeId::UINTEGER:
+	case LogicalTypeId::HUGEINT:
+	case LogicalTypeId::BIGINT:
+	case LogicalTypeId::FLOAT:
+	case LogicalTypeId::DOUBLE:
+	case LogicalTypeId::DECIMAL:
+	case LogicalTypeId::UUID:
+	case LogicalTypeId::BLOB: // TODO: check BLOB and BIT 'binary' primitive type ?
```

Comment:
BIT is not supported, it doesn't map to an Iceberg type

Reply:
True, I was thinking we could cast unsigned to long or something, but forgot about implementing the rest of that. It can be done later. 
Although I should probably add a test to make sure we error out when trying to write a UINT column

Reply:
Hmm, already covered by the `create table as select ... from test_all_types()`